core/main/src/service/apps/delegated_launcher_handler.rs

```diff
@@ -489,80 +489,151 @@ impl DelegatedLauncherHandler {
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
             (Paused, Active) => {
-                self.emit_lifecycle_app_event(&app_id, OnActivate, old_state, new_state, None)
-                    .await;
+                Self::emit_lifecycle_app_event(
+                    platform_state,
+                    &app_id,
+                    OnActivate,
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
             (Active, Paused) => {
-                self.emit_lifecycle_app_event(&app_id, OnPause, old_state, new_state, None)
-                    .await;
+                Self::emit_lifecycle_app_event(
+                    platform_state,
+                    &app_id,
+                    OnPause,
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
             (Paused, Suspended) => {
-                self.emit_lifecycle_app_event(&app_id, OnSuspend, old_state, new_state, None)
-                    .await;
+                Self::emit_lifecycle_app_event(
+                    platform_state,
+                    &app_id,
+                    OnSuspend,
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
             (Suspended, Paused) => {
-                self.emit_lifecycle_app_event(&app_id, OnResume, old_state, new_state, None)
-                    .await;
+                Self::emit_lifecycle_app_event(
+                    platform_state,
+                    &app_id,
+                    OnResume,
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
             (Suspended, Hibernated) => {
-                self.emit_lifecycle_app_event(&app_id, OnHibernate, old_state, new_state, None)
-                    .await;
+                Self::emit_lifecycle_app_event(
+                    platform_state,
+                    &app_id,
+                    OnHibernate,
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
             (Hibernated, Suspended) => {
-                self.emit_lifecycle_app_event(&app_id, OnRestore, old_state, new_state, None)
-                    .await;
+                Self::emit_lifecycle_app_event(
+                    platform_state,
+                    &app_id,
+                    OnRestore,
+                    old_state,
+                    new_state,
+                    None,
+                )
+                .await;
             }
-            (Paused, Terminating) => {
+            (_, Terminating) => {
```

Comment:
Did we change the state transition logic? 
As per the HLA, transition to `Terminating` is allowed only from `Paused` state.